applications/Preprocessor/lib/mesh/MeshImpl/MeshEntity_Impl.h

```diff
@@ -145,13 +145,89 @@ std::vector<EntityGId> MeshEntity<
     }
 }
 
+template <std::size_t entityDimension, std::size_t meshDimension>
+template <std::size_t d>
+std::enable_if_t<d == entityDimension>
+    MeshEntity<entityDimension, meshDimension>::merge(
+        MeshEntity<entityDimension, meshDimension>& source, tag<d>) {
+    // Merging elements is fundamentally more complicated and should not be done
+    // via this method.
+    static_assert(entityDimension < meshDimension,
+                  "Can't use this to merge elements");
+
+    logger.assert_debug(source.getMesh() == getMesh(),
+                        "Merging from different meshes");
+    logger.assert_debug(this->getGlobalIndex() != source.getGlobalIndex(),
+                        "Can't merge an entity into itself");
+    // Loop to move all connections
+    // Internally removes the links from the source element.
+    while (!source.elementIDs.empty()) {
+        Element<meshDimension>& element = source.getElement(0);
+        EntityLId localId = source.getLocalIndex(0);
+        if (entityDimension != 0) {
+            element.template setEntity<entityDimension>(localId,
+                                                        getGlobalIndex());
+        }
+    }
+}
+
```

Comment:
hmmm... maybe use a private template function to remove most of the duplicate code?